### PR TITLE
Binding issues [test_ops]

### PR DESCRIPTION
**Pull request by athul-bos-semi**
### Ticket
Link to Github Issue #10650 

### Problem description

I am trying to bind an Op to TTNN, and followed the steps shown in the [guide](https://tenstorrent.github.io/tt-metal/latest/ttnn/ttnn/adding_new_ttnn_operation.html). But I am facing the following error:
```
In file included from ../ttnn/cpp/pybind11/__init__.cpp:14:
In file included from ../ttnn/cpp/pybind11/operations/__init__.hpp:29:
../ttnn/cpp/ttnn/operations/test_ops/test_ops_pybind.hpp:14:44: error: no member named 'test_concat' in namespace 'ttnn'
   14 |     detail::bind_test_concat(module, ttnn::test_concat); 
      |                                      ~~~~~~^
1 error generated.
ninja: build stopped: subcommand failed.

```

### What's changed
1. Created the directory ttnn/cpp/ttnn/operations/test_ops, and added all files inside it
2. In ttnn/cp/pybind11/operation/__init__.hpp, I added the following two lines inside operations namespace
```
auto m_test = module.def_submodule("test_ops", "Test operations");
test_ops::py_module(m_test);
```
3. Added the file ttnn/ttnn/operations/test_concat.py with the following code:
```
from typing import Tuple, Union, Dict

import sys
import ttnn

test_concat = ttnn.register_operation()(ttnn._ttnn.operations.test_ops.test_concat)

__all__ = []
```

### CODE REVIEW

File: ttnn/ttnn/operations/test_concat.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# SPDX-FileCopyrightText: © 2024 BOS
+
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Tuple, Union, Dict
+
+import sys
+import ttnn
+
```
Comment (ayerofieiev-tt) [inline ttnn/ttnn/operations/test_concat.py:9]: This file is not needed unless you want to register a golden function

File: ttnn/cpp/ttnn/operations/test_ops/Concat/concat_device_operation.hpp [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    static ttnn::Shape compute_output_shapes(const operation_attributes_t&, const tensor_args_t&);
+
+    static Tensor create_output_tensors(const operation_attributes_t& operation_attributes, const tensor_args_t&);
+
+};
+
+}
+}
+constexpr auto test_concat = ttnn::test_concat<operations::test_ops::Concat>("ttnn::test_concat");
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/test_ops/Concat/concat_device_operation.hpp:86]: This registers a wrong structure as an operation. 
there are two structures which play different roles. 

Operation and DeviceOperation

the folder structure is 

* ttnn
  * operations
    * category (in this case can be experimental)
      * <operation> (name should be lowered, snake case like test_concat)
        * <operation>.hpp (contains the op - simple structure which defines op interface via operator() and calls register ttnn operation)
        * <operation>_pybind.hpp
        * device (optional, if you have device kernels)
          * <operation>_device_operation.hpp/.cpp (contains device op struct which defines how to validate args, how to calculate output shape, how to create output tensor, how to setup program/kernels)
          * <operation>_progam_factory
          * kernels
            * …
        * host (optional, if you have a host implementation)
          * …

Comment (ayerofieiev-tt) [reply ttnn/cpp/ttnn/operations/test_ops/Concat/concat_device_operation.hpp:86]: looks like you missed this step
https://tenstorrent.github.io/tt-metal/latest/ttnn/ttnn/adding_new_ttnn_operation.html#step-2-implement-the-operation-in-c

**Comment by athul-bos-semi** [comment]
It seems to me the format given in https://tenstorrent.github.io/tt-metal/latest/ttnn/ttnn/adding_new_ttnn_operation.html is different from the format of new Ops from the current main branch. For example, dividing Ops into Single and multicore have been done away with, and variables inside Ops are now const variables instead of being saved inside structs named operation_attributes and tensor_args.
Which format do you recommend we follow?

**Pull request closed without merging**